src/java/org/apache/cassandra/repair/autorepair/FixedSplitTokenRangeSplitter.java

```diff
@@ -102,9 +102,11 @@ private KeyspaceRepairAssignments getRepairAssignmentsForKeyspace(boolean primar
         boolean byKeyspace = config.getRepairByKeyspace(repairType);
         // collect all token ranges.
         List<Range<Token>> allRanges = new ArrayList<>();
+        // this is done to avoid micro splits in the case of vnodes
+        int splitsPerRange = Math.max(1, numberOfSubranges / tokens.size());
```

Comment:
Can we adjust `number_of_subranges` default as well (line 72) while we are at it?  I think we can safely adjust this to 64 now that we are splitting this way, but I think we discussed 32 which I'd be comfortable with as well :+1: